tasty-query/shared/src/main/scala/tastyquery/Symbols.scala

```diff
@@ -668,6 +677,65 @@ object Symbols {
         throw MemberNotFoundException(this, name)
       }
 
+    /** Returns a list of all the overloaded declarations with the given unsigned name.
+      *
+      * If there is no declaration with the given unsigned name, this method
+      * returns `Nil`.
+      *
+      * @throws java.lang.IllegalArgumentException
+      *   if the provided `name` is a [[SignedName]]
+      */
+    final def getAllOverloadedDecls(name: TermName)(using Context): List[TermSymbol] =
+      name match
+        case name: SignedName => throw IllegalArgumentException(s"Illegal SignedName argument: $name")
+        case _                => myDeclarations.get(name).fold(Nil)(_.toList.map(_.asTerm))
+    end getAllOverloadedDecls
+
+    /** Returns a list of all the overloaded declarations with the given unsigned name.
+      *
+      * @throws tastyquery.Exceptions.MemberNotFoundException
+      *   if there is no declaration with the given unsigned name
+      * @throws java.lang.IllegalArgumentException
+      *   if the provided `name` is a [[SignedName]]
+      */
+    final def findAllOverloadedDecls(name: TermName)(using Context): List[TermSymbol] =
+      getAllOverloadedDecls(name) match
+        case Nil   => throw MemberNotFoundException(this, name)
+        case decls => decls
+    end findAllOverloadedDecls
+
+    /** Convenience method to get a non-overloaded decl from its unsigned name.
+      *
+      * If there are multiple or no overload with the given unsigned name, this
+      * method returns `None`.
+      *
+      * @throws java.lang.IllegalArgumentException
+      *   if the provided `name` is a [[SignedName]]
+      */
+    final def getNonOverloadedDecl(name: TermName)(using Context): Option[TermSymbol] =
```

Comment:
I guess this isn't statically SimpleName for convenience?

Reply:
There are other `TermName`s that are not `SignedName`s, for example `DefaultGetterName`s.